workspace/src/main/kotlin/com/cosmotech/workspace/service/WorkspaceServiceImpl.kt

```diff
@@ -106,56 +109,77 @@ internal class WorkspaceServiceImpl(
     return result
   }
 
-  override fun findWorkspaceById(organizationId: String, workspaceId: String): Workspace {
+  override fun getWorkspace(organizationId: String, workspaceId: String): Workspace {
     return updateSecurityVisibility(getVerifiedWorkspace(organizationId, workspaceId))
   }
 
-  override fun createWorkspace(organizationId: String, workspace: Workspace): Workspace {
+  override fun createWorkspace(organizationId: String, workspaceCreateRequest: WorkspaceCreateRequest): Workspace {
     organizationService.getVerifiedOrganization(
         organizationId, listOf(PERMISSION_READ, PERMISSION_CREATE_CHILDREN))
     // Validate Solution ID
-    workspace.solution.solutionId?.let { solutionService.findSolutionById(organizationId, it) }
+    workspaceCreateRequest.solution.solutionId.let { solutionService.findSolutionById(organizationId, it) }
 
+    val workspaceId = idGenerator.generate("workspace")
+    val security = csmRbac.initSecurity(
+      workspaceCreateRequest.security.toGenericSecurity(workspaceId)).toResourceSecurity()
     val createdWorkspace =
-        workspace.copy(
-            id = idGenerator.generate("workspace"),
-            organizationId = organizationId,
-            ownerId = getCurrentAuthenticatedUserName(csmPlatformProperties))
-    createdWorkspace.setRbac(csmRbac.initSecurity(workspace.getRbac()))
+      Workspace(
+        id = workspaceId,
+        organizationId = organizationId,
+        ownerId = getCurrentAuthenticatedUserName(csmPlatformProperties),
+        key = workspaceCreateRequest.key,
+        name = workspaceCreateRequest.name,
+        solution = workspaceCreateRequest.solution,
+        security = security,
+        version = workspaceCreateRequest.version,
+        tags = workspaceCreateRequest.tags,
+        description = workspaceCreateRequest.description,
+        webApp = workspaceCreateRequest.webApp,
+        datasetCopy = workspaceCreateRequest.datasetCopy,
+        )
 
     return workspaceRepository.save(createdWorkspace)
   }
 
-  override fun deleteAllWorkspaceFiles(organizationId: String, workspaceId: String) {
+  override fun deleteWorkspaceFiles(organizationId: String, workspaceId: String) {
     val workspace = getVerifiedWorkspace(organizationId, workspaceId, PERMISSION_WRITE)
     deleteAllFilesystemWorkspaceFiles(organizationId, workspace)
   }
 
   override fun updateWorkspace(
-      organizationId: String,
-      workspaceId: String,
-      workspace: Workspace
+    organizationId: String,
+    workspaceId: String,
+    workspaceUpdateRequest: WorkspaceUpdateRequest
   ): Workspace {
     val existingWorkspace = this.getVerifiedWorkspace(organizationId, workspaceId, PERMISSION_WRITE)
     // Security cannot be changed by updateWorkspace
+    val updatedWorkspace = Workspace(
```

Comment:
It's still buggy here. 
All WorkspaceUpdateRequest optional fields should be tested.
If not tags/webApp/datasetCopy/... is set in `WorkspaceUpdateRequest`, all fields will be reset to null.
Please add test with empty `WorkspaceUpdateRequest` to cover this correctly